src/WebJobs.Script.WebHost/App_Start/WebHostResolver.cs

```diff
@@ -43,45 +43,60 @@ public WebHostResolver(ScriptSettingsManager settingsManager, ISecretManagerFact
 
         public ILoggerFactory GetLoggerFactory(WebHostSettings settings)
         {
-            WebScriptHostManager manager = GetWebScriptHostManager(settings);
-
-            if (!manager.CanInvoke())
+            // if we have an active initialized host, return it's fully configured
+            // logger factory
+            var hostManager = GetWebScriptHostManager(settings);
+            if (hostManager.CanInvoke())
             {
-                // The host is still starting and the LoggerFactory cannot be used. Return
-                // an empty LoggerFactory rather than waiting. This will mostly affect
-                // calls to /admin/host/status, which do not block on CanInvoke. It allows this
-                // API to remain fast with the side effect of occassional missed logs.
-                return _emptyLoggerFactory;
+                var config = GetScriptHostConfiguration(settings);
+                return config.HostConfig.LoggerFactory;
+            }
+            else
+            {
+                // if there is no active host, return the default logger factory
+                if (_defaultLoggerFactory == null)
+                {
+                    lock (_syncLock)
+                    {
+                        if (_defaultLoggerFactory == null)
+                        {
+                            var config = GetScriptHostConfiguration(settings);
+                            var clientFactory = new ScriptTelemetryClientFactory(_settingsManager.ApplicationInsightsInstrumentationKey, config.ApplicationInsightsSamplingSettings, config.LogFilter.Filter);
+                            _defaultLoggerFactory = new LoggerFactory();
+                            _defaultLoggerFactory.AddApplicationInsights(clientFactory);
+                        }
+                    }
+                }
+                
+                return _defaultLoggerFactory;
             }
-
-            return GetScriptHostConfiguration(settings).HostConfig.LoggerFactory;
         }
 
         public TraceWriter GetTraceWriter(WebHostSettings settings)
         {
-            // if we have an active host, return it's fully configured
+            // if we have an active initialized host, return it's fully configured
             // trace writer
             var hostManager = GetWebScriptHostManager(settings);
-            var traceWriter = hostManager.Instance?.TraceWriter;
```

Comment:
The issue was that we had an Instance, but it wasn't fully initialized so its TraceWriter was only set to the default SystemTraceWriter which wasn't logging to file.